public/scripts/app.js

```diff
@@ -2,6 +2,502 @@ console.log("Sanity Check: JS is working!");
+
+  // opens up the 4 text fields
+  $('.input-for-profile-header').toggle();
+  // removes the 'at' inbetween Title and workPlace
+  $('.at-span').toggle();
+  // removes the name in the h1
+  $('.h1-name').toggle();
+  // removes all the info from spans that is from the DB
+  $('.profile-header-db-info').toggle();
+  // shows the table under each of the input / text box area
+  $('.label-profile-header').toggle();
+  // hides the edit-header pencil that is currently showing and gets rid of that while showing the new one
+  $('.edit-input-header').toggle();
+  $('.edit-save-header').toggle();
+  // hides the upload file icon
+  $('.modal-input-for-file').toggle();
+
+
+};
+
+// ability save all the edits made to the profile header section and rerender that profile on the SRP
+let saveProfileHeader = function(e) {
+  console.log('SAVE HEADER ICON WAS CLICKED')
+  let currentProfileId = $(this).closest('#profile-information').attr('data-profile-id');
+  console.log('SAVE ICON CLICKED AND THIS IS THE PROFILE ID', currentProfileId)
+
+
+  // get the profileHeader label and once after you get that
+  let editedProfileHeader = $(this).closest('#profile-information');
+
+  // go and get each of the value from the input field. The initial value of each field was populated automatically by using the 'value=' in the input tag of the html
+  let editedProfileHeaderName = editedProfileHeader.find('.input-for-profile-name').val();
+  let editedProfileHeaderTitle = editedProfileHeader.find('.input-for-profile-title').val();
+  let editedProfileHeaderWorkpalce = editedProfileHeader.find('.input-for-profile-workPlace').val();
+  let editedProfileHeaderQuote = editedProfileHeader.find('.textarea-for-profile-quote').val();
+  let editProfileHeaderImg ='/images/userimages/' +  $('.input-for-file').val().split('\\').pop();
+  console.log('THIS IS EDIT PROFILE IMAGE',editProfileHeaderImg )
+
+  // create the data object to send back
+  let editedProfileHeaderData = {
+    name: editedProfileHeaderName,
+    title: editedProfileHeaderTitle,
+    image: editProfileHeaderImg,
+    workPlace: editedProfileHeaderWorkpalce,
+    quote: editedProfileHeaderQuote,
+  };
+  console.log('THIS IS THE DATA TO BE SENT BACK AFTER EDIT', editedProfileHeaderData);
+
+  $.ajax({
+    method: 'PUT',
+    url: `/api/searchpage/${currentProfileId}`,
+    data: editedProfileHeaderData,
+  })
+  .then(function(updatedEditedData) {
+    console.log('DATA returned from editedProfileHeaderData', updatedEditedData);
+    // send back the updated data do can pass it through a similar render function again. This time will have another function b/c one is coming from the SRP and the other is coming from the profile page
+    renderProfileAfterEdit(updatedEditedData);
+  })
+  .catch(function(err) {
+    console.log('ERROR during the editedProfileHeaderData returned data', err);
+  });
+  // closes the save icon
+  $('.edit-save-header').toggle();
```

Comment:
Give one class to all of those elements called ".toggle" and then just take everything with that class and toggle it.